### ISSUE TITLE
Module in Menu not working if not Assigned to a Page

### ISSUE THREAD

**Issue by DamirPecnik** [gantry/gantry5#147]
When module is in menu if its assigned to a page it will show in menu on that page if its not it will not show, don't know if this is a bug or intentional but would bee good if this is not page based or maybe make a option show only on assigned pages or show site wide that could be good to.

**Comment by mahagr** [comment]
We are using Joomla methods and rules to render modules. Same rules apply when you try to render module in a component; if its not assigned to a page (menu item), it will not render.

To be clear, Joomla has 2 kinds of assignments for modules:

* Assign to module position (Module tab)
* Assign to menu items (Menu Assignment tab)

Module position can be None, but you need to always select menu items or pages, you want the module to show up -- that is just how Joomla wants to do it.

I would rather do it in Joomla way, because of otherwise we need to copy all that code from Joomla into Gantry and update the code every single time Joomla makes changes to it.

Basically right now our code works just like it was designed to work; our code does exactly the same as it would do when you add ```<jdoc:include type="modules" name="my-position" style="gantry" />``` or ```<jdoc:include type="module" name="mymodule" style="gantry" />``` into your template/layout file.

Well, the later one is slightly more powerful as it takes real instance of the module and also allows frontend editing.
